context/src/junit/bindings.rs

```diff
@@ -31,6 +32,178 @@ pub struct BindingsReport {
+            TestCaseRunStatus::Failure => BindingsTestCaseStatusStatus::NonSuccess,
+            TestCaseRunStatus::Skipped => BindingsTestCaseStatusStatus::Skipped,
+            TestCaseRunStatus::Unspecified => todo!(),
+        }
+    }
+}
+
+impl From<TestResult> for BindingsReport {
+    fn from(
+        TestResult {
+            test_case_runs,
+            uploader_metadata,
+        }: TestResult,
+    ) -> Self {
+        let test_cases: Vec<BindingsTestCase> = test_case_runs
+            .into_iter()
+            .map(|testcase| BindingsTestCase::from(testcase))
+            .collect();
+        let parent_name_map: HashMap<String, Vec<BindingsTestCase>> =
+            test_cases.iter().fold(HashMap::new(), |mut acc, testcase| {
+                let parent_name = testcase.extra.get("parent_name").unwrap();
+                acc.entry(parent_name.clone())
+                    .or_insert_with(Vec::new)
+                    .push(testcase.to_owned());
+                acc
+            });
+        let mut report_time = 0.0;
+        let mut report_failures = 0;
+        let mut report_tests = 0;
+        let test_suites = parent_name_map
+            .into_iter()
+            .map(|(name, testcases)| {
+                let tests = testcases.len();
+                let disabled = testcases
+                    .iter()
+                    .filter(|tc| tc.status.status == BindingsTestCaseStatusStatus::Skipped)
+                    .count();
+                let failures = testcases
+                    .iter()
+                    .filter(|tc| tc.status.status == BindingsTestCaseStatusStatus::NonSuccess)
+                    .count();
+                let timestamp = testcases.iter().map(|tc| tc.timestamp.unwrap_or(0)).max();
+                let timestamp_micros = testcases
+                    .iter()
+                    .map(|tc| tc.timestamp_micros.unwrap_or(0))
+                    .max();
+                let time = testcases.iter().map(|tc| tc.time.unwrap_or(0.0)).sum();
+                report_time += time;
+                report_failures += failures;
+                report_tests += tests;
+                BindingsTestSuite {
+                    name,
+                    tests,
+                    disabled,
+                    errors: 0,
+                    failures,
+                    timestamp,
+                    timestamp_micros,
+                    time: Some(time),
+                    test_cases: testcases,
+                    properties: vec![],
+                    system_out: None,
+                    system_err: None,
+                    extra: HashMap::new(),
+                }
+            })
+            .collect();
+        let (name, timestamp, timestamp_micros) = match uploader_metadata {
+            Some(t) => (
+                t.origin,
+                Some(t.upload_time.unwrap_or_default().seconds),
+                Some(t.upload_time.unwrap_or_default().nanos as i64 * 1000),
```

Comment:
millis * 1000?

Reply:
1000 microseconds is a nanosecond

Reply:
Unless I'm blind, I'm reading this as

`timestamp_micros = upload_time.nanos * 1000` (instead of / 1000). As written this would give you picoseconds